rstar/src/algorithm/iterators.rs

```diff
@@ -43,10 +43,31 @@ where
     }
 }
 
-impl<'a, T, Func> Iterator for SelectionIterator<'a, T, Func>
+pub fn select_nodes<'a, T, Func, Cons>(root: &'a ParentNode<T>, func: &Func, cons: &mut Cons)
 where
     T: RTreeObject,
+    Func: SelectionFunction<T>,
+    Cons: FnMut(&'a T) -> bool,
+{
+    if func.should_unpack_parent(&root.envelope()) {
+        for node in root.children.iter().rev() {
```

Comment:
I used `.rev()` to stay exactly compatible with the internal stack-based implementation, but I do wonder whether this is actually necessary?

Reply:
Nah, the order of elements should be considered random in general. Let's not bother with that.

Reply:
Done.